src/backend/common/CommandBuffer.cpp

```diff
@@ -472,6 +594,28 @@ namespace backend {
                             return false;
                         }
 
+                        // TODO(kainino@chromium.org): We should be able to
+                        // optimize this by tracking which textures are in use
+                        // as attachments. Maybe it's possible that the
+                        // XAttachment usages could mark this: disallow
```

Comment:
note: This could work and this would mean the end usage will have to be encoded in the renderpass. I'm not 100% sure this is desirable but don't see a counter example.